lib/ext/adviser.rb

```diff
@@ -0,0 +1,19 @@
+require 'csv'
+
+module AdviserExtension
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def to_csv(_ = {})
+      CSV.generate do |csv|
+        csv << ['Ref. Number', 'Name', 'Firm']
+
+        find_each do |adviser|
+          csv << [adviser.reference_number, adviser.name, adviser.firm.registered_name]
+        end
+      end
+    end
+  end
+end
+
+Adviser.send(:include, AdviserExtension)
```

Comment:
With the wrapper you manipulate (or you can call extend in a gross way) *without modifying the model*. This opens ways to add more features without the *need to modify the model or touch the to_csv method*.  Imho, this respects the open/closed principle. There is a link on the wiki page on the PR review that @margOnline created that would be interested for you to read: https://maswiki.valiantyscloud.net/display/NewDev/Development+workflow
It is called "Checklist for reviewing code" that there is a small part on open closed principle.